@here/harp-mapview/lib/geometry/DisplacedMesh.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { DisplacementFeature, hasDisplacementFeature } from "@here/harp-materials";
+import { assert } from "@here/harp-utils";
+import * as THREE from "three";
+import { DisplacedBufferAttribute } from "./DisplacedBufferAttribute";
+import { DisplacedBufferGeometry, DisplacementRange } from "./DisplacedBufferGeometry";
+
+function isBufferGeometry(
+    geometry: THREE.Geometry | THREE.BufferGeometry
+): geometry is THREE.BufferGeometry {
+    const isBufferGeom = geometry instanceof THREE.BufferGeometry;
+    assert(isBufferGeom, "Unsupported geometry type.");
```

Comment:
This assertions inside the check function looks little strange :). Either way you assert also from caller. Please remove all asserts out of isXXX function scope.

Reply:
I'm only asserting when calling from the constructor, not when calling from raycast. The issue there is that I need to keep an if so that tslint applies the casting correctly to the input parameters (material , geometry and displacementMap) so that they are recognized by tslint as having the appropiate types.
Anyway, I'll remove the assert from the constructor call.